### PR TITLE
Add k8s client support for service resources

### PR DESCRIPTION
**Pull request by jsravn**

### CODE REVIEW

**Comment by jsravn** [comment]
This is ready for review.

File: controller/controller.go [hunk 1]
```diff
@@ -67,29 +67,19 @@ func (c *controller) Start() error {
...
-			return
-		case <-c.watcher.Updates():
-			log.Info("Received update on watcher")
-			err := c.updateLoadBalancer()
-			if err != nil {
-				log.Errorf("Unable to update load balancer: %v", err)
-			}
+	for _ = range c.watcher.Updates() {
+		log.Info("Received update on watcher")
+		err := c.updateLoadBalancer()
```
Comment (balooo) [inline controller/controller.go]: can we rename 'updateLoadBalancer'  to just 'update' or 'doUpdate', but it's not load balancer specific anymore

File: controller/controller.go [hunk 2]
```diff
@@ -67,29 +67,19 @@ func (c *controller) Start() error {
...
+		if err != nil {
+			log.Errorf("Unable to update load balancer: %v", err)
```
Comment (balooo) [inline controller/controller.go]: I don't think this is load balancer specific

**Comment by balooo** [comment]
lgtm, 2 minor comments

**Pull request merged**